File: 3434.maximum-frequency-after-subarray-operation.java
#
# @lc app=leetcode id=3434 lang=java
#
# [3434] Maximum Frequency After Subarray Operation
#

# @lc code=start
class Solution {
    public int maxFrequency(int[] nums, int k) {
        int n = nums.length;
        int maxFreq = 0;
        for (int i = 0; i < n; ++i) {
            int cnt = 0;
            // Try to expand to the left
            for (int l = i; l >= 0; --l) {
                int x = k - nums[l];
                boolean valid = true;
                for (int m = l; m <= i; ++m) {
                    if (nums[m] + x != k) {
                        valid = false;
                        break;
                    }
                }
                if (valid) cnt = Math.max(cnt, i - l + 1);
            }
            maxFreq = Math.max(maxFreq, cnt);
        }
        // Also count original k's
        int freqK = 0;
        for (int num : nums) if (num == k) freqK++;
        return Math.max(maxFreq, freqK);
    }
}
# @lc code=end

File: 3434.maximum-frequency-after-subarray-operation.go
#
# @lc app=leetcode id=3434 lang=golang
#
# [3434] Maximum Frequency After Subarray Operation
#

# @lc code=start
func maxFrequency(nums []int, k int) int {
    // Sort nums to allow efficient evaluation of subarrays
    sort.Ints(nums)
    left := 0
    sum := 0 // To store sum of elements within current window
    maxFreq := 0 // To store maximum frequency found
    // Iterate over nums with right as end of sliding window
    for right := 0; right < len(nums); right++ {
        sum += nums[right] // Add current element to sum
        // Check if we can transform current window into nums[right]
        while nums[right]*(right-left+1)-sum > k {
            sum -= nums[left] // Remove leftmost element from sum
            left++ // Move left pointer forward
        }
        // Update max frequency if current window is valid
        maxFreq = max(maxFreq, right-left+1)
    }
    return maxFreq // Return maximum frequency found
}
// Helper function to determine max of two integers
func max(a, b int) int { if a > b { return a } else { return b }}
# @lc code=end